src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMNavigator.java

```diff
@@ -237,8 +239,35 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
         }
         for (BitbucketRepository repo : repositories) {
             checkInterrupt();
-            add(listener, observer, repo);
+            if (shouldObserveRepo(listener, repo)) {
+                add(listener, observer, repo);
+            }
+        }
+    }
+
+    private boolean shouldObserveRepo(TaskListener listener, BitbucketRepository repo) {
+        if (!isAutoRegisterHooks()) {
+            return true;
+        }
+        String name = repo.getRepositoryName();
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
+            boolean jobExists = jenkins.getItems().stream()
```

Comment:
Oh god no, this is not the way to address this issue:

1. It assumes OrganizationFolder
2. It is crawling all items
3. You will never be able to have repos that no longer match be orphaned out

If it is a rate limit issue then we just need to use sleeping until reset logic like in GitHub Branch Source